internal/cmd/connect/command_plugin_install.go

```diff
@@ -0,0 +1,456 @@
+	}
+
+	return ownerNameSplit[0], nameVersionSplit[0], nameVersionSplit[1], nil
+}
+
+func getLocalManifest(archivePath string) (*manifest, error) {
+	zipReader, err := zip.OpenReader(archivePath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to open local archive file %s", archivePath)
+	}
+	defer zipReader.Close()
+
+	for _, zipFile := range zipReader.File {
+		isManifest, err := filepath.Match("*/manifest.json", filepath.ToSlash(zipFile.Name))
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to examine file %s inside local archive file %s", zipFile.Name, archivePath)
+		}
+
+		if isManifest {
+			manifestFile, err := zipFile.Open()
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to open manifest file %s inside local archive file %s", zipFile.Name, archivePath)
+			}
+			defer manifestFile.Close()
+
+			jsonByteArr, err := io.ReadAll(manifestFile)
+			if err != nil {
+				return nil, err
+			}
+
+			pluginManifest := new(manifest)
+			if err := json.Unmarshal(jsonByteArr, &pluginManifest); err != nil {
+				return nil, err
+			}
+
+			return pluginManifest, nil
+		}
+	}
+
+	return nil, errors.Errorf(`failed to find manifest file inside local archive file "%s"`, archivePath)
+}
+
+func getRemoteManifest(owner, name, version string) (*manifest, error) {
+	manifestUrl := fmt.Sprintf("https://api.hub.confluent.io/api/plugins/%s/%s", owner, name)
+	if version != "latest" {
+		manifestUrl = fmt.Sprintf("%s/versions/%s", manifestUrl, version)
+	}
+
+	r, err := http.Get(manifestUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.StatusCode != http.StatusOK {
+		return nil, errors.New("failed to read manifest file from Confuent Hub")
+	}
+
+	defer r.Body.Close()
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	pluginManifest := new(manifest)
+	if err := json.Unmarshal(body, &pluginManifest); err != nil {
+		return nil, err
+	}
+
+	return pluginManifest, nil
+}
+
+func getPluginDirFromFlag(cmd *cobra.Command) (string, error) {
+	if cmd.Flags().Changed("plugin-directory") {
+		pluginDir, err := cmd.Flags().GetString("plugin-directory")
+		if err != nil {
+			return "", err
+		}
+
+		if pluginDir, err = filepath.Abs(pluginDir); err != nil {
+			return "", err
+		}
+
+		if !utils.DoesPathExist(pluginDir) {
+			return "", errors.Errorf(invalidDirectoryErrorMsg, pluginDir)
+		}
+
+		return pluginDir, nil
+	}
+
+	return "", nil
+}
+
+func existingPluginInstallation(pluginDir string, pluginManifest *manifest) ([]string, error) {
+	// Bundled installations
+	if utils.DoesPathExist(filepath.Join(pluginDir, pluginManifest.Name)) {
+		return nil, errors.New("unable to install plugin because it is already bundled")
+	}
+
+	// Other previous installations
+	immediateDirectory := filepath.Join(pluginDir, fmt.Sprintf("%s-%s", pluginManifest.Owner.Username, pluginManifest.Name))
+	uberJar := filepath.Join(pluginDir, fmt.Sprintf("%s-%s.jar", pluginManifest.Name, pluginManifest.Version))
```

Comment:
Uber? What does that mean?

Reply:
I have no idea. That's what they're called in the original code so I just carried it over.

Reply:
Did you reach out to the Connect team yet? Maybe someone there can clarify this for you? It would be helpful to write down what this means in a comment.

Reply:
I reached out again a few minutes ago.